WordPress/Sniffs/Theme/DiscouragedFunctionsSniff.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Discouraged functions.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_DiscouragedFunctionsSniff extends WordPress_AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Groups of functions to restrict.
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type'      => 'error' | 'warning',
+	 * 		'message'   => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'site_url' => array(
+				'type'      => 'warning',
+				'message'   => '%s() found. Use home_url() instead.',
```

Comment:
The wording is a bit hard. What about "%s() found. In most cases home_url() is a better function to use."?

The reason I say this is that reviewers may not see that it is `Warning` and start requiring it even though it may be a valid use.